src/org/zaproxy/zap/extension/ascanrulesAlpha/TestNoSQLInjection.java

```diff
@@ -0,0 +1,505 @@
+import org.zaproxy.zap.model.TechSet;
+
+import net.sf.json.JSONException;
+import net.sf.json.JSONObject;
+
+
+/**
+ * @author l.casciaro
+ *
+ */
+public class TestNoSQLInjection extends AbstractAppParamPlugin {
+	
+	
+	
+	// Prefix for internationalised messages used by this rule
+	 
+	private static final String MESSAGE_PREFIX = "ascanalpha.testnosqlinjection.";
+	private static final String JSON_ATTACK = "json";
+	private static final String PARVAL_ATTACK = "paramvalue";
+	private static final String VALUE_ATTACK = "value";
+	private static final String CRASH_ATTACK = "crash";
+	private static final String TIMED_ATTACK ="timed";
+	private static final String TOKEN ="OWASP_ZAP_TOKEN_INJECTION";
+	private static final int SLEEP_TIME_SHORT=500;
+	private static final int SLEEP_TIME_LONG=2000;
+	private static final int DELTA_TIME=200;
+
+	private static final int ID = 40033;
+	private static final int CWED_ID = 943;
+	private static final int WASC_ID = 19;
+	
+	private static final String MONGO_INDEX = "mongo";
+	private static final String COUCH_INDEX = "couch";
+	private String indexDbSel = MONGO_INDEX;
+	
+	
+	
+	//START MongoDB
+	private static final String[] MONGO_ERROR_MATCHING_STRING = 
+
+			new String[] { 
+					"retval", "Unexpected token", "mongoDB", "SyntaxError", "ReferenceError", "Illegal", TOKEN };
+	
+	private static final String[][] MONGO_URL_PARAM_VALUE_INJECTION = new String[][] {
+		{ "[$ne]", "0"}, {"[$regex]", ".*"}, {"[$gt]", "0"}};
+		
+	private static final String[] MONGO_URL_ERROR_INJECTION = 	{
+		"\"",
+		"'",
+		"//",
+	};
+	
+	private static final String[] MONGO_URL_VALUE_INJECTION = {
+		"'; return (true); var notReaded='",
+		"'); print("+TOKEN+"); print('",
+		"_id);}, function(kv) { return 1; },"	+
+		"{ out: 'x'}); print('Injection'); "	+ 
+		"return 1; db.noSQL_injection.mapReduce(function()" + 
+		"{ emit(1,1",
+		"true, $where: '1 == 1'",		
+	};
+
+	private static final String[][] MONGO_TIMED_VALUE_INJECTION = {
+			{"\"'); sleep("+SLEEP_TIME_SHORT+"); print('\"", "\"'); sleep("+SLEEP_TIME_LONG+"); print('\""},
+			{"'; sleep("+SLEEP_TIME_SHORT+"); var x='", "'; sleep("+SLEEP_TIME_LONG+"); var x='"}
+	};
+	
+	private static final String[][] MONGO_JSON_PARAM_VALUE_INJECTION = { 
+		{"$ne", "0"},
+		{"$gt", ""},
+		{"$regex", ".*"}
+	};
+	// END MongoDB
+	
+	// START CouchDB
+	private static final String[] COUCH_URL_VALUE_INJECTION = {
+			"_all_dbs",
+			"_changes",
+			"_compact",
+			"_design",
+			"temp_view",
+			"_view_cleanup",
+			"_info",
+			"_list",
+			"_rewrite",
+			"_show",
+			",_update",
+			"_view",
+			"_acrive_tasks"
+	};
+	// END CouchDB
+	
+	
+	private static final String[][] EMPTY_PARAM_VALUE = new String[0][0];
+	private static final String[] EMPTY_VALUE = new String[0];	
+	
+	// The set of all NoSQL DB-Drivers technology pairs to test
+	private enum NOSQLDB{
+
+		mongoDB(Tech.MongoDB, MONGO_URL_VALUE_INJECTION, MONGO_URL_ERROR_INJECTION, MONGO_URL_PARAM_VALUE_INJECTION,

```

Comment:
The new noSQL Tech entries can't be used until the new core is moved over, which won't happen until just before 2.8.0. You'll have to find a way to work around that for the time being. Perhaps just leave them as Tech.Db for the time being?

This is what's causing the builds to fail.